### PR TITLE
HPCC-33116 Suppress file check if invalid @compressedSize(0)

### PR DESCRIPTION
**Pull request by jakesmith**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: dali/base/dadfs.cpp [hunk 1]
```diff
@@ -13807,6 +13807,39 @@ void configurePreferredPlanes()
     }
 }
 
+static bool doesPhysicalMatchMeta(IPropertyTree &partProps, IFile &iFile, offset_t &expectedSize, offset_t &actualSize)
```
Comment (ghalliday) [inline dali/base/dadfs.cpp:13810]: trivial: expectedSize should not be a reference for this function.

File: dali/base/dadfs.cpp [hunk 2]
```diff
@@ -13807,6 +13807,39 @@ void configurePreferredPlanes()
...
+{
+    // NB: temporary workaround for 'narrow' files publishing extra empty parts with the wrong @compressedSize(0)
+    // causing a new check introduced in HPCC-33064 to be hit (fixed in HPCC-33113, but will continue to affect exiting files)
+    unsigned __int64 size = partProps.getPropInt64("@size", (unsigned __int64)-1);
+    unsigned __int64 compressedSize = partProps.getPropInt64("@compressedSize", (unsigned __int64)-1);
+    if ((0 == size) && (0 == compressedSize))
+        return true;
```
Comment (ghalliday) [inline dali/base/dadfs.cpp:13817]: actualSize should be set to 0 (or unknownFileSize)

File: dali/base/dadfs.cpp [hunk 3]
```diff
@@ -13807,6 +13807,39 @@ void configurePreferredPlanes()
     }
 }
 
+static bool doesPhysicalMatchMeta(IPropertyTree &partProps, IFile &iFile, offset_t &expectedSize, offset_t &actualSize)
+{
+    // NB: temporary workaround for 'narrow' files publishing extra empty parts with the wrong @compressedSize(0)
+    // causing a new check introduced in HPCC-33064 to be hit (fixed in HPCC-33113, but will continue to affect exiting files)
+    unsigned __int64 size = partProps.getPropInt64("@size", (unsigned __int64)-1);
```
Comment (ghalliday) [inline dali/base/dadfs.cpp:13814]: Picky: could use unknownFileSize

**Review by ghalliday — COMMENTED** [review]
@jakesmith a few picky comments. None are actually bugs.

**Comment by jakesmith** [comment]
@ghalliday - please see 2nd commit.

**Review by ghalliday — APPROVED** [review]
Please squash

**Comment by jakesmith** [comment]
@ghalliday - squashed.

**Pull request merged**